packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -144,9 +142,16 @@ export class SharedTreeCore<
 	) {
 		super(id, runtime, attributes, telemetryContextPrefix);
 
-		this.stableId = uuid();
-		editManager.initSessionId(this.stableId);
-		this.indexes = isReadonlyArray(indexes) ? indexes : indexes(this.indexEventEmitter);
+		/**
+		 * A random ID that uniquely identifies this client in the collab session.
+		 * This is sent alongside every op to identify which client the op originated from.
+		 * This is used rather than the Fluid client ID because the Fluid client ID is not stable across reconnections.
+		 */
+		const localSessionId = uuid();
```

Comment:
There's a sketchy cast going on here. If the session ID is going to use the SessionId type, then it should also use the SessionId creator instead of an any conversion.

Reply:
Right, this isn't the IdCompressor SessionId, it's just a type that has been in EditManager for a while. I've been shuffling it around as I do these refactors, but yes I'm planning on replacing it with the IdCompressor one when that is possible.